pkg/cmd/roachtest/roachtestutil/utils.go

```diff
@@ -163,9 +165,25 @@ func CloseExporter(
 	}
 }
 
+// WaitForSQLReady waits until the corresponding node's SQL subsystem is fully initialized and ready
+// to serve SQL clients.
+// N.B. The fact that we have a live db connection doesn't imply that the SQL subsystem is ready to serve. E.g.,
+// a SQL session cannot be authenticated until after `SyntheticPrivilegeCache` is initialized, which is done
+// asynchronously at server startup.
+// (See "Root Cause" in https://github.com/cockroachdb/cockroach/issues/137988)
+func WaitForSQLReady(ctx context.Context, db *gosql.DB) error {
+	retryOpts := retry.Options{MaxRetries: 5}
+	return retryOpts.Do(ctx, func(ctx context.Context) error {
+		_, err := db.ExecContext(ctx, "SELECT 1")
+		return err
+	})
+}
+
 // WaitForReady waits until the given nodes report ready via health checks.
 // This implies that the node has completed server startup, is heartbeating its
 // liveness record, and can serve SQL clients.
+// FIXME(srosenberg): This function is a bit of a misnomer. It doesn't actually ensure that SQL is ready to serve, only
```

Comment:
@andrewbaptist Just a heads-up since I believe this helper is used only by the tests you authored. While the Admin endpoint may be up, it doesn't guarantee that SQL is ready. The current implementation may be "ok" for these tests, but it's more pronounced in shared-process, multi-tenant configurations.